src/tools/miri/src/shims/panic.rs

```diff
@@ -220,6 +220,34 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriInterpCxExt<'mir, 'tcx> {
                     },
                 )?;
             }
+            MisalignedPointerDereference { required, found } => {
+                // Forward to `panic_misaligned_pointer_dereference` lang item.
```

Comment:
Please add a test that actually emits this panic -- probably in `tests/pass/panic/catch_panic.rs`.

Reply:
This PR contains a UI test for the panic. Does the `catch_panic` test cover something else?

Reply:
I meant a test for Miri. One taht actually exercises this path in this file.